.github/workflows/provider-publish-service-artifacts.yml

```diff
@@ -0,0 +1,100 @@
+name: Provider Publish Service Artifacts
+
+on:
+  workflow_call:
+    inputs:
+      subpackages:
+        description: 'Subpackages to be built individually (e.g. monolith config ec2)'
+        required: true
+        type: string
+    secrets:
+      UPBOUND_MARKETPLACE_PUSH_ROBOT_USR:
+        required: true
+      UPBOUND_MARKETPLACE_PUSH_ROBOT_PSW:
+        required: true
+
+env:
+  # Common versions
+  GO_VERSION: '1.19'
+  GOLANGCI_VERSION: 'v1.50.0'
+  DOCKER_BUILDX_VERSION: 'v0.8.2'
+
+  # Common users. We can't run a step 'if secrets.XXX != ""' but we can run a
+  # step 'if env.XXX' != ""', so we copy these to succinctly test whether
+  # credentials have been provided before trying to run steps that need them.
+  UPBOUND_MARKETPLACE_PUSH_ROBOT_USR: ${{ secrets.UPBOUND_MARKETPLACE_PUSH_ROBOT_USR }}
+  
+jobs:  
+  publish-service-artifacts:
+    if: ${{ github.actor == 'OWNER' || github.actor == 'turkenf' || contains(github.event.repository.collaborators, github.actor) }}
```

Comment:
`github.actor` designates the username of the Github user who initially ran the workflow:
https://docs.github.com/en/actions/learn-github-actions/contexts#github-context

Because we currently do not make any membership role checks with the [current CI workflow](https://github.com/upbound/uptest/blob/main/.github/workflows/provider-ci.yml), I suggest that we also do not do any extra authorization checks here. In future iterations, we may consider further restricting who can trigger this new workflow or the existing CI workflow based on the membership roles of the Github users:
```suggestion
```